Source/BugsnagSessionTracker.m

```diff
@@ -116,11 +117,21 @@ - (void)startNewSessionWithAutoCaptureValue:(BOOL)isAutoCaptured {
         return;
     }
 
-    self.currentSession = [[BugsnagSession alloc] initWithId:[[NSUUID UUID] UUIDString]
-                                                   startDate:[NSDate date]
-                                                        user:self.config.user
-                                                autoCaptured:isAutoCaptured];
+    BugsnagSession *newSession = [[BugsnagSession alloc] initWithId:[[NSUUID UUID] UUIDString]
+                                                          startDate:[NSDate date]
+                                                               user:self.config.user
+                                                       autoCaptured:isAutoCaptured];
+    // TODO JL: refactor to pass as structured data into callback
+    BugsnagSessionTrackingPayload *payload = [[BugsnagSessionTrackingPayload alloc] initWithSessions:@[]
+                                                                                              config:self.config];
+    NSMutableDictionary *data = [payload toJson];
+    for (BugsnagOnSessionBlock cb in self.config.onSessionBlocks) {
+        if (!cb(data)) {
+            return;
+        }
+    }
```

Comment:
nit: this refactoring is a good opportunity to improve naming for clarity.  What do you think about:

* `payload` -> `sessions`
* `data` -> `sessionsJson`
* `cb` -> `onSessionBlock`

Reply:
sounds good :) renamed